pkg/resources/oauth_integration_for_custom_clients.go

```diff
@@ -0,0 +1,666 @@
+		DeleteContext: DeleteContextOauthIntegrationForCustomClients,
+
+		CustomizeDiff: customdiff.All(
+			ComputedIfAnyAttributeChanged(
+				ShowOutputAttributeName,
+				"name",
+				"enabled",
+				"comment",
+			),
+			ComputedIfAnyAttributeChanged(
+				DescribeOutputAttributeName,
+				"oauth_client_type",
+				"oauth_redirect_uri",
+				"enabled",
+				"oauth_allow_non_tls_redirect_uri",
+				"oauth_enforce_pkce",
+				"oauth_use_secondary_roles",
+				"pre_authorized_roles_list",
+				"blocked_roles_list",
+				"oauth_issue_refresh_tokens",
+				"oauth_refresh_token_validity",
+				"network_policy",
+				"oauth_client_rsa_public_key",
+				"oauth_client_rsa_public_key_2",
+				"comment",
+			),
+		),
+
+		Importer: &schema.ResourceImporter{
+			StateContext: ImportOauthForCustomClientsIntegration,
+		},
+	}
+}
+
+func ImportOauthForCustomClientsIntegration(ctx context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+	logging.DebugLogger.Printf("[DEBUG] Starting oauth integration for custom clients import")
+	client := meta.(*provider.Context).Client
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+
+	integration, err := client.SecurityIntegrations.ShowByID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	integrationProperties, err := client.SecurityIntegrations.Describe(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = d.Set("enabled", strconv.FormatBool(integration.Enabled)); err != nil {
+		return nil, err
+	}
+
+	if allowNonTlsRedirectUri, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+		return property.Name == "OAUTH_ALLOW_NON_TLS_REDIRECT_URI"
+	}); err == nil {
+		if err = d.Set("oauth_allow_non_tls_redirect_uri", allowNonTlsRedirectUri.Value); err != nil {
+			return nil, err
+		}
+	}
+
+	if enforcePkce, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+		return property.Name == "OAUTH_ENFORCE_PKCE"
+	}); err == nil {
+		if err = d.Set("oauth_enforce_pkce", enforcePkce.Value); err != nil {
+			return nil, err
+		}
+	}
+
+	if issueRefreshTokens, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+		return property.Name == "OAUTH_ISSUE_REFRESH_TOKENS"
+	}); err == nil {
+		if err = d.Set("oauth_issue_refresh_tokens", issueRefreshTokens.Value); err != nil {
+			return nil, err
+		}
+	}
+
+	if refreshTokenValidity, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+		return property.Name == "OAUTH_REFRESH_TOKEN_VALIDITY"
+	}); err == nil {
+		refreshTokenValidityValue, err := strconv.ParseInt(refreshTokenValidity.Value, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		if err = d.Set("oauth_refresh_token_validity", refreshTokenValidityValue); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Because import is a step before read, so the rest it read in the Read itself, but the question is do we want to do it this way or make import "complete" with all of the fields.

Reply:
Hmm I thought we were doint "complete" imports, but if it works, it's fine with me. So, if read is called anyway, we could set only id/name, right?